gulpfile.js/build.js

```diff
@@ -392,10 +463,404 @@ function buildPages(done) {
+      4
+    ),
+    newPost(
+      'Scattered houses in a mountain village.',
+      'landscape_village_1280x853.jpg',
+      5
+    ),
+    newPost('A deep canyon.', 'landscape_canyon_1280x1700.jpg', 6),
+    newPost(
+      'A desert with mountains in the background.',
+      'landscape_desert_1280x853.jpg',
+      7
+    ),
+    newPost('Colorful houses on a street.', 'landscape_houses_1280x803.jpg', 8),
+    newPost('Blue sea surrounding a cave.', 'landscape_sea_1280x848.jpg', 9),
+    newPost(
+      'A ship sailing the sea at sunset.',
+      'landscape_ship_1280x853.jpg',
+      10
+    ),
+  ];
+
+  const configObj = {
+    time: new Date().toLocaleTimeString(),
+    timestamp: Number(new Date()),
+    // send a random list of blog items to make it also work on the cache
+    blogItems: blogItems.filter(() =>
+      Math.floor(Math.random() * Math.floor(2))
+    ),
+  };
+
+  return gulp
+    .src(`${project.paths.DIST}/examples/sources/**/*.html`)
+    .pipe(
+      through.obj(async (file, enc, callback) => {
+        const srcHTML = file.contents.toString();
+
+        env.renderString(srcHTML, configObj, (err, result) => {
+          if (err) {
+            logger.error(`Error rendering ${file.path}`);
+            return callback(err);
+          }
+
+          file.contents = Buffer.from(result);
+          callback(null, file);
+        });
+      })
+    )
+    .pipe(gulp.dest((f) => f.base))
+    .on('end', () => {
+      done();
+    });
+}
+
+/**
+ * Compiles the pages into standalone static files
+ *
+ * @return {Promise}
+ */
+async function staticify(done) {
+  const logger = require('@lib/utils/log')('Static File Generator');
+
+  const requestPathRegex = new RegExp(
+    `${project.paths.PAGES_DEST}|(index)?.html`,
+    'g'
+  );
+
+  const generatedFormats = SUPPORTED_FORMATS.map((format) => {
+    const f = (cb) => {
+      const env = nunjucksEnv();
+
+      const f = filter([`${project.paths.PAGES_DEST}/**/*html`, 'index'], {
+        restore: true,
+      });
+
+      return gulp
+        .src(`${project.paths.PAGES_DEST}/**/*.html`)
+        .pipe(f)
+        .pipe(
+          // Render a static version of all of our pages
+
+          through.obj(async (file, enc, callback) => {
+            const configObj = {
+              requestPath: `${file.path.replace(requestPathRegex, '')}/`,
+              format,
+              requestedFormat: format,
+            };
+
+            const srcHTML = file.contents.toString();
+
+            env.renderString(srcHTML, configObj, (err, result) => {
+              if (err) {
+                logger.error(`Error rendering ${file.path}`);
+                return callback(err);
+              }
+
+              file.contents = Buffer.from(result);
+              callback(null, file);
+            });
+          })
+        )
+        .pipe(
+          through.obj(async function (file, enc, callback) {
+            // Rewrite links inside of each of the pages
```

Comment:
This rewrites only "known" format specific links, but format specific links in docs etc won't be rewritten, but rely on redirects - correct?